zookeeper-server/src/main/java/org/apache/zookeeper/cli/SyncCommand.java

```diff
@@ -55,14 +57,18 @@ public CliCommand parse(String[] cmdArgs) throws CliParseException {
     @Override
     public boolean exec() throws CliException {
         String path = args[1];
+        CountDownLatch latch = new CountDownLatch(1);
```

Comment:
What about a more modern CompletableFuture?

Reply:
Thanks for mentioning this good option:`CompletableFuture`,but for issue,IMO,we don't need to kill a chicken with the sword：D

Reply:
`CompletableFuture` has specifically been designed for scenarios like this. As far as I'm concerned. Let me check.

Reply:
How about this?

```java
    @Override
    public boolean exec() throws CliException {
        String path = args[1];
        CompletableFuture<Integer> cf = new CompletableFuture<>();

        try {
            zk.sync(path, new AsyncCallback.VoidCallback() {
                public void processResult(int rc, String path, Object ctx) {
                    cf.complete(rc);
                }
            }, null);

            try {
                int resultCode = cf.get(CONNECTION_TIMEOUT, TimeUnit.MILLISECONDS);
                out.println("Sync returned " + resultCode);
            } catch (TimeoutException ex) {
                out.println("Sync is timeout within " +  CONNECTION_TIMEOUT + " ms");
            }
        } catch (IllegalArgumentException | InterruptedException | ExecutionException ex) {
            throw new MalformedPathException(ex.getMessage());
        }

        return false;
    }
```

Reply:
@anmolnar  Great work. you are so sweet!  it's very elegant especially when we want to transfer the result value.
Cc:Thanks @eolivelli  for mentioning this good option:`CompletableFuture` again.